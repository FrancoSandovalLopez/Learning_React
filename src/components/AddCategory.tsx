import { FormEvent, SetStateAction, useState } from "react";

type Props = {
    //setCategories: (value: SetStateAction<string[]> ) => void, 
    onNewCategory: (value: string ) => void
}

export const AddCategory = ({ onNewCategory }:Props) => {

    const [inputValue, setInputValue] = useState('');

    const onSubmit = (e: FormEvent) => {
        e.preventDefault();
        if(inputValue.trim().length <= 1) return;
        // setCategories((x: string[]): string[] => {
        //     return [inputValue, ...x];
        // });
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    const onAddInput = (value: string) => {
        setInputValue(value);
    }

    return (
        <form onSubmit={(e) => onSubmit(e)}>
            <input
                type="text"
                placeholder="Search Gifs"
                value={inputValue}
                onChange={(e) => onAddInput(e.target.value)}
            />
        </form>
    );
}
