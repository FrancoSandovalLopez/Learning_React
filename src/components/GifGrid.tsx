
type Props = {
    category: string
}

export const GifGrid = ( {category}: Props ) => {
  return (
    <>
        <h3>{ category }</h3>   
    </>
  )
}
