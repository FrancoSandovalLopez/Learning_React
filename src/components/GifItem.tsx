
interface gif {
    id: string,
    title: string,
    url: string
}

type Props = {
    url: string
    title: string
}

export const GifItem = (image: Props) => {

    console.log(image);

    return (
        <div className="card">
            <img src={image.url} alt={image.title} />
            <p>{image.title}</p>
        </div>
    )
}
