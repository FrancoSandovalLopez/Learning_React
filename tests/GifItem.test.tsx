import React from 'react';
import { render, screen } from '@testing-library/react';
import { GifItem } from '../src/components/GifItem';

describe('GifItem component', () => {
  const props = {
    url: 'https://example.com/image.gif',
    title: 'Example image',
  };

  test('should render the image and title', () => {
    const { container } = render(<GifItem {...props} />);
    expect( container ).toMatchSnapshot(); 
  });
});