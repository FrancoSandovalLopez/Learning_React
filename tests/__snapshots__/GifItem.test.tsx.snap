// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component should render the image and title 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Example image"
      src="https://example.com/image.gif"
    />
    <p>
      Example image
    </p>
  </div>
</div>
`;
